crates/core/tedge_api/src/entity_store.rs

```diff
@@ -483,7 +483,12 @@ impl EntityStore {
 
             Ok(register_messages)
         } else {
-            Err(Error::NonDefaultTopicScheme(entity_topic_id.clone()))
+            debug!(
+                "Failed to create auto register message due to: {}",
```

Comment:
We should log that something "failed" as we are just saying that the topic being used does not trigger an auto registration messages (which is expected).

Maybe something like this would be more accurate/helpful to the user:

```suggestion
                "Topic does not match auto registration scheme: {}",
```

Reply:
I'll leave the review of whether the error should be bubble up to the calling function up to the Rust reviewers.

Reply:
This won't be enough. The error still bubbles up all the way up to the `convert` function which will log that error and publish it to `tedge/errors` as well. It's better to catch this error from the caller: `try_convert_te_topics` and do the logging there and avoid bubbling it up any further.

Reply:
Now I have `short circuted` and returns an error message on the `te/errors` message. if the schema is not supported by auto-registration.